Use the supplied contact name in SesEmailComposer

mvelez-nydig opened the pull request:
Possible fix for https://github.com/micronaut-projects/micronaut-email/issues/117

nmostowich-nydig commented on email-amazon-ses/src/test/groovy/io/micronaut/email/ses/SesEmailComposerSpec.groovy:
```diff
@@ -110,4 +110,28 @@ class SesEmailComposerSpec extends Specification {
         !request.destination().ccAddresses()
     }
+
+    void "from field should allow to include the sender name"() {
```
nit but should say "allow including" rather than "allow to include"

sdelamo commented on email-amazon-ses/src/main/java/io/micronaut/email/ses/SesEmailComposer.java:
```diff
@@ -136,4 +136,14 @@ private Body.Builder bodyBuilder(@NonNull Email email) {
+    @NonNull
+    private String sourceFormatter(@NonNull Email email) {
+        io.micronaut.email.Contact from = email.getFrom();
+        if (from.getName() != null && !from.getName().isEmpty()) {
```
Please use https://docs.micronaut.io/latest/api/io/micronaut/core/util/StringUtils.html#isNotEmpty-java.lang.CharSequence-

sdelamo commented on email-amazon-ses/src/main/java/io/micronaut/email/ses/SesEmailComposer.java:
```diff
@@ -136,4 +136,14 @@ private Body.Builder bodyBuilder(@NonNull Email email) {
     }
+
+    @NonNull
+    private String sourceFormatter(@NonNull Email email) {
```
make the method protected to allow override and add javadoc

sdelamo commented on email-amazon-ses/src/test/groovy/io/micronaut/email/ses/SesEmailComposerSpec.groovy:
```diff
@@ -110,4 +110,28 @@ class SesEmailComposerSpec extends Specification {
+
+    void "from field should allow including the sender name"() {
+        given:
+        def from = new Contact("sender@example.com", "John Doe")
```
```suggestion
        Contact from = new Contact("sender@example.com", "John Doe")
```

sdelamo commented on email-amazon-ses/src/test/groovy/io/micronaut/email/ses/SesEmailComposerSpec.groovy:
```diff
@@ -110,4 +110,28 @@ class SesEmailComposerSpec extends Specification {
+    void "from field should allow including the sender name"() {
+        given:
+        def from = new Contact("sender@example.com", "John Doe")
+        def formattedFrom = "${from.getName()} <${from.getEmail()}>"
```
```suggestion
        String formattedFrom = "${from.getName()} <${from.getEmail()}>"
```

sdelamo commented on email-amazon-ses/src/test/groovy/io/micronaut/email/ses/SesEmailComposerSpec.groovy:
```diff
@@ -110,4 +110,28 @@ class SesEmailComposerSpec extends Specification {
+        given:
+        def from = new Contact("sender@example.com", "John Doe")
+        def formattedFrom = "${from.getName()} <${from.getEmail()}>"
+        def to = "receiver@example.com"
```
```suggestion
        String to = "receiver@example.com"
```

sdelamo commented on email-amazon-ses/src/test/groovy/io/micronaut/email/ses/SesEmailComposerSpec.groovy:
```diff
@@ -110,4 +110,28 @@ class SesEmailComposerSpec extends Specification {
+        def from = new Contact("sender@example.com", "John Doe")
+        def formattedFrom = "${from.getName()} <${from.getEmail()}>"
+        def to = "receiver@example.com"
+        def subject = "Apple Music"
```
```suggestion
        String subject = "Apple Music"
```

sdelamo commented on email-amazon-ses/src/test/groovy/io/micronaut/email/ses/SesEmailComposerSpec.groovy:
```diff
@@ -110,4 +110,28 @@ class SesEmailComposerSpec extends Specification {
+                .body("Lore ipsum body")
+                .build()
+        when:
+        def request = sesEmailComposer.compose(email) as SendEmailRequest
```
```suggestion
        SendEmailRequest request = sesEmailComposer.compose(email) as SendEmailRequest
```

sdelamo reviewed (changes requested):
Please, the base branch of this PR should be 1.4.x. Can you change it?

mvelez-nydig commented:
> Please, the base branch of this PR should be 1.4.x. Can you change it?

@sdelamo changed the base branch and addressed your comments. Let me know if these look right
